addons/link_tracker/models/link_tracker.py

```diff
@@ -197,6 +196,27 @@ def search_or_create(self, vals):
 
         return self.create(vals)
 
+    @api.model
+    def search_or_create_multi(self, vals_list):
+        for vals in vals_list:
+            if 'url' not in vals:
+                raise ValueError(_('Creating a Link Tracker without URL is not possible'))
+            vals['url'] = tools.validate_url(vals['url'])
+
+        search_domains = [[
+            (fname, '=', value)
+            for fname, value in vals.items()
+            if fname in {'url', 'campaign_id', 'medium_id', 'source_id', 'label'}
+        ] for vals in vals_list]
+
+        search_domain = expression.OR(search_domains)
+        result = self.search(search_domain)
+
+        if not result:
+            return self.create(vals_list)
```

Comment:
Technically you also have to check for duplicates inside ``vals_list``(a search or create is always more complicated in multi, this is why people often prefer the sequential one :D ) . You could have no existing trackers, but duplicates in the given list.